### PR TITLE
Break out S3 bucket settings to align with `aws` provider 4.x changes

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This pull request does the following:
1. Update bucket name generation to be saner.
2. Break out S3 bucket setting attributes into their own corresponding resources.
3. Change the bucket access permissions to reject all forms of public access by default.

## 💭 Motivation and context ##

We need to make changes in support of the `aws` provider 4.x version that have been backported into the 3.x version to support migrating our base provider version to 4.x as a development team. The name generation for the bucket names was overly complicated so they were updated. Lastly both buckets may contain sensitive data so it's best to block public access as a default action.

## 🧪 Testing ##

Automated tests pass. This has been applied in my test environment (and apparently a production environment) without issue.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

File: exported_data_bucket.tf [hunk 1]
```diff
@@ -1,24 +1,41 @@
+# This bucket is used to store a JSON file containing all exported
+# findings data to be imported into our AWS database.
```
Comment (dav3r) [inline exported_data_bucket.tf]: I think this is a bit more accurate, don't you?
```suggestion
# This bucket is used to store JSON files containing exported
# assessment findings data to be imported into our AWS database.
```

File: lambda_bucket.tf [hunk 2]
```diff
@@ -1,28 +1,41 @@
+# This bucket is used to store the lambda function that imports findings
```
Comment (dav3r) [inline lambda_bucket.tf]: I believe we have been capitalizing this:
```suggestion
# This bucket is used to store the Lambda function that imports findings
```

**Review by dav3r — APPROVED** [review]
Excellent work. 👍 👍 
I noted a couple of minor comment edits.

File: exported_data_bucket.tf (hunk 1 above)
Comment (mcdonnnj) [reply exported_data_bucket.tf]: Please see [a5e0343](https://github.com/cisagov/findings-data-import-terraform/pull/36/commits/a5e0343952066821f98ec6a0a99d140fd4442d28).

File: lambda_bucket.tf (hunk 2 above)
Comment (mcdonnnj) [reply lambda_bucket.tf]: Please see [99e1c8c](https://github.com/cisagov/findings-data-import-terraform/pull/36/commits/99e1c8c73f9c91be8c3efeebed29e18000f2b0bd).

**Review by jsf9k — APPROVED** [review]

**Pull request merged**